modules/javafx.web/src/main/native/Source/WebCore/page/DOMWindow.cpp

```diff
@@ -845,22 +845,18 @@ ExceptionOr<Storage*> DOMWindow::localStorage()
         return Exception { SecurityError };
 
     auto* page = document->page();
-    // FIXME: We should consider supporting access/modification to local storage
-    // after calling window.close(). See <https://bugs.webkit.org/show_bug.cgi?id=135330>.
-    if (!page || !page->isClosing()) {
-        if (m_localStorage)
-            return m_localStorage.get();
-    }
-
     if (!page)
         return nullptr;
 
-    if (page->isClosing())
-        return nullptr;
-
+    // Check if localstorage setting is disabled, then return nullptr
     if (!page->settings().localStorageEnabled())
         return nullptr;
 
+    // FIXME: We should consider supporting access/modification to local storage
+    // after calling window.close(). See <https://bugs.webkit.org/show_bug.cgi?id=135330>.
+    if (page->isClosing() && m_localStorage)
+            return m_localStorage.get();
+
```

Comment:
If you make the earlier modification I suggested, then you don't need this block.

Reply:
We must check  if localstorage setting is disabled, then return nullptr first.,as below
    if (!page->settings().localStorageEnabled())
        return nullptr;

and there after the section 
    if (page->isClosing() && m_localStorage)
            return m_localStorage.get();
would become as below
    if (m_localStorage)
            return m_localStorage.get();

Reply:
in recent code base (webkit upgrade)the localstorage is false ,
@@ -1579,7 +1579,7 @@ NeedsSiteSpecificQuirks:
     WebKit:
       default: true
     WebCore:
-      default: false
+      default: true

So , it needs to be enable also.

Reply:
yes, in case of not enable , in any case localstorage would be null

Reply:
That's what I would have guessed, but since it seems to be working well, I'm puzzled. Are there cases where local storage is disabled when it should be enabled?